### PR TITLE
add backup and deletion protection examples

### PR DESCRIPTION
**Pull request by ykethan**
#### Description of changes:
adds examples on enabling deletion protection on DynamoDB tables created by Amplify Data GraphQL API and Cognito user pool created by Amplify Auth
Enable point-in-time recovery for DynamoDB tables
Create scheduled backups for DynamoDB tables using AWS backup service.

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/[platform]/build-a-backend/add-aws-services/deletion-backup-resources/index.mdx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+```ts title="amplify/backend.ts"
+const dataResources = backend.data.resources.cfnResources;
+for (const table of Object.values(dataResources.amplifyDynamoDbTables)) {
+  table.pointInTimeRecoveryEnabled = true;
+}
+```
+
+## Enabling Backups for DynamoDB tables
+
+For example, if your DynamoDB tables requires backups that extend the default 35 days Point-in-time recovery, AWS Backup service can be utilized to centralize and automate backups for DynamoDB tables.
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/add-aws-services/deletion-backup-resources/index.mdx:85]:
```suggestion
For example, if your DynamoDB tables requires backups that extend the default 35 days point-in-time recovery, AWS Backup service can be utilized to centralize and automate backups for DynamoDB tables. 
```

File: src/pages/[platform]/build-a-backend/add-aws-services/deletion-backup-resources/index.mdx [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+const backend = defineBackend({
+  auth,
+  data,
+});
+
+const backupStack = backend.createStack("backup-stack");
+const ITables = Object.values(backend.data.resources.tables);
+
+const vault = new BackupVault(backupStack, "backup-vault", {
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/add-aws-services/deletion-backup-resources/index.mdx:109]:
```suggestion
const vault = new BackupVault(backupStack, "BackupVault", {
```

File: src/pages/[platform]/build-a-backend/add-aws-services/deletion-backup-resources/index.mdx [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  backupVaultName: "backup-vault",
+});
+
+const plan = new BackupPlan(backupStack, "backup-plan", {
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/add-aws-services/deletion-backup-resources/index.mdx:113]:
```suggestion
const plan = new BackupPlan(backupStack, "BackupPlan", {
```

File: src/pages/[platform]/build-a-backend/add-aws-services/deletion-backup-resources/index.mdx [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+      minute: "0",
+      hour: "0",
+      day: "*",
+      month: "*",
+      year: "*",
+    }),
+  })
+);
+
+plan.addSelection("plan-selection", {
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/add-aws-services/deletion-backup-resources/index.mdx:132]:
```suggestion
plan.addSelection("BackupPlanSelection", {
```

assuming this is a logical ID

**Review by timngyn — APPROVED** [review]

**Review by timngyn — DISMISSED** [review]

File: src/pages/[platform]/build-a-backend/add-aws-services/deletion-backup-resources/index.mdx [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+  data
+});
+```
+
+## Enabling Deletion protection on a Auth resource
+
+For example, if you would like to enable Deletion protection on a Cognito user pool resource created by Amplify Auth.
+
+```ts title="amplify/backend.ts"
+backend.auth.resources.cfnResources.cfnUserPool.deletionProtection = "ACTIVE";
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/add-aws-services/deletion-backup-resources/index.mdx:58]:
```suggestion
import { defineBackend } from '@aws-amplify/backend';
import { auth } from './auth/resource';
import { data } from './data/resource';

const backend = defineBackend({
  auth,
  data
});

// highlight-start
const { cfnUserPool } = backend.auth.resources.cfnResources
cfnUserPool.deletionProtection = "ACTIVE";
// highlight-end
```

nit for copy-pastable example

File: src/pages/[platform]/build-a-backend/add-aws-services/deletion-backup-resources/index.mdx [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+```
+
+## Enabling Deletion protection on a Data resource
+
+For example, if you would like to enable Deletion protection on all DynamoDB tables created by GraphQL API.
+
+```ts title="amplify/backend.ts"
+const dataResources = backend.data.resources.cfnResources;
+for (const table of Object.values(dataResources.amplifyDynamoDbTables)) {
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/add-aws-services/deletion-backup-resources/index.mdx:67]:
```suggestion
const { amplifyDynamoDbTables } = backend.data.resources.cfnResources;
for (const table of Object.values(amplifyDynamoDbTables)) {
```

for readability

separately, is this located under cfnResources?

File: src/pages/[platform]/build-a-backend/add-aws-services/deletion-backup-resources/index.mdx [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+import { auth } from "./auth/resource";
+import { data } from "./data/resource";
+
+const backend = defineBackend({
+  auth,
+  data,
+});
+
+const backupStack = backend.createStack("backup-stack");
+const ITables = Object.values(backend.data.resources.tables);
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/add-aws-services/deletion-backup-resources/index.mdx:107]: can we choose a different name for this? the `I<something>` convention is typically reserved for interfaces

File: src/pages/[platform]/build-a-backend/add-aws-services/deletion-backup-resources/index.mdx (hunk 6 above)
Comment (ykethan) [reply src/pages/[platform]/build-a-backend/add-aws-services/deletion-backup-resources/index.mdx:67]: yes, this is now located under cfnResources: 
https://docs.amplify.aws/gen2/build-a-backend/data/override-resources/

**Review by josefaidt — APPROVED** [review]

**Review by timngyn — APPROVED** [review]

**Pull request merged**